mbed_host_tests/host_tests_conn_proxy/conn_primitive_fastmodel.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+"""
+mbed SDK
+Copyright (c) 2011-2018 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import telnetlib
+import socket
+from .conn_primitive import ConnectorPrimitive, ConnectorPrimitiveException
+
+
+class FastmodelConnectorPrimitive(ConnectorPrimitive):
+    def __init__(self, name, config):
+        ConnectorPrimitive.__init__(self, name)
+        self.config          = config
+        self.fm_config      = config.get('fm_config', None)
+        self.platform_name   = config.get('platform_name', None)
+        self.image_path      = config.get('image_path', None)
+        self.polling_timeout = int(config.get('polling_timeout', 60))
+
+        # FastModel Agent tool-kit
+        self.fm_agent_module = None
+        self.resource = None
+
+        # Initialize FastModel
+        if self.__fastmodel_init():
+        
+            # FastModel Launch load and run, equivalent to DUT connection, flashing and reset...
+            self.__fastmodel_launch()
+            self.__fastmodel_load(self.image_path)
+            self.__fastmodel_run()
+
+
+    def __fastmodel_init(self):
+        """! Initialize models using fm_agent APIs """
+        self.logger.prn_inf("Initializing FastModel...")
+
+        try:
+            self.fm_agent_module = __import__("fm_agent")
```

Comment:
If we're going to know the name of the python module and we intend to release the python module on pip, can we go ahead and just add this as a dependency? The python module itself doesn't have any heavy dependencies correct?

Do you have an opinion here @theotherjimmy?

Reply:
My opinion is exactly that.

Reply:
I wouldn't suggest put the `fm_agent` module as a dependency to htrun for now, because:
1. `fm_agent` module do have another dependecy module, which is closed source. So I don't think it will be published in pip
2. `fm_agent`  module has some manual configuration steps need to be done by the users before it can be functional.
3. Not every user would be interested in using fast models. some people would like just use greentea/htrun with their own hardware, so they don't bother about these thing. htrun can used as it was.

Reply:
1. Fair point, though if `fm_agent` is included by default at least now they only have to worry about one "optional" module.
2. That's good to know, we may want to revist this. If the info is simple enough, we may be able to use an environment variable OR we can make it part of the greentea/htrun CLI.
3. Again, I wouldn't worry too much about this. If it isn't a heavy dependency and it doesn't break the existing interface then I don't think this is a problem. It can just be there and not do any harm 😄

Reply:
Hi guys, @bridadan @theotherjimmy 
I am back from a week vacation. So is there a conclusion on this matter yet?

I think the fundamental question is, do we really want htrun tightly bound with fm_agent module?  htrun will not run without it (even for people have no interest in using fast models ).

Or do we want to decouple them? So htrun will work as it was, on normal targets. evern fm_agent is not present.

Reply:
Hi @jamesbeyond, thanks for following up. @theotherjimmy and I were just discussing this offline.

What we determined was that since `fm_agent` is dependent on the commercial `fm` module (not installable through the pypi registry) that comes with the FastModel product, then I think what we'll want to do is conditionally expose the `--fm` option in both htrun and greentea based on if the `fm` module is installed/importable.

Perhaps this functionality can be added to `fm_agent` and greentea and htrun can just call into that module to see if everything is installed? Not sure of the specifics here, I'd be curious to hear your thoughts as well as @theotherjimmy's.

Reply:
Hi @bridadan, thanks for getting back so quick.

I think the current changes do exactly what you said: 
the fm_agent module checks those required depends of fast-models to be installed.
htrun just call to use the modules.

BTW, This 2 PRs have been there for some time, I am really like to get them merged before 5.9 close up. Do you think there is enough time for it? or maybe we should speed it up a little bit?

Or Is this PR will still require other stakeholders to review or more CIs to pass? this is my first PR to the tools, I am not sure about the procedures yet.

Reply:
@jamesbeyond Not quite. I would like the `--fm` switch to only be present when the fm_agent module is installed. As this PR stands, the `--fm` switch is always available.

Reply:
Haa! I see the point.
That's not a big change. I will get that in ASAP.

Any comments on the timing to merge?

Reply:
10 min after that patch!

Reply:
@jamesbeyond I think I was saying something different than @theotherjimmy, sorry for the confusion! We synced offline and I wanted to post those conclusions here just so we're sure we're on the same page:

1. Let's leave `fastmodel_agent` as an optional install, just like you do now. Since it requires the extra configuration step and the `fm` module, we'll leave that up to the user to take care of before using htrun/greentea.
2. We would like the `--fm` switch to only be enabled when `fastmodel_agent` is installed/importable. So the `parser.add_option('', '--fm', ...` call will need to be conditional. This is to reduce users exposure to this unless they have `fastmodel_agent` installed and configured.

Reply:
Thanks to both of you for the clarification. didn't expect the merge is that quick ;-)